src/run/handlers/cache.cts

```diff
@@ -393,7 +394,7 @@ export class NetlifyCacheHandler implements CacheHandlerForMultipleVersions {
 
       await this.cacheStore.set(key, { lastModified, value }, 'blobStore.set')
 
-      if (data?.kind === 'PAGE' || data?.kind === 'PAGES') {
+      if (!data || data?.kind === 'PAGE' || data?.kind === 'PAGES') {
```

Comment:
This might be too broad, we really should try to figure out exact key that is being invalidated as a result of calling `res.revalidate` and use that as condition instead of inspecting `data` at all.

We also can't just store argument passed to `res.revalidate` as-is, because Next.js might do some massaging to it so the argument might not match cache key later (somewhat similar to why we can't use request pathname to generate cache tags reliably).

This is quite tricky and I don't think we do have good tools today to figure this out correctly, but we will need to find some to do it

Reply:
Updated in 4cf7b2f400a31d1a067ca8e129fae04106cf3070